### ISSUE TITLE
[Support]: lower than expected performance

### ISSUE THREAD

**Issue by Narolth** [blakeblackshear/frigate#3393]
### Describe the problem you are having

I'll start by explaining that I work in IT, but primarily on the windows side, so the performance issue or a lack of understanding on my side is entirely possible. 

I have 4x HP DL 380 servers, running esx 7.0.3, vsphere, and fully configured in my home lab. I provisioned an ubuntu virtual machine, installed Portainer, and then completed the frigate install using `docker compose up` after some review of documentation, I have an instance up and working, using wyze rtsp cameras. I have 12 camera's on my iot network, and the VM has a nic in that network to communicate directly with the devices etc. 

I am not using a coral device, as I have multiple esx hosts and I do not want to pin the virtual machine running docker to a host for usb passthrough. 

I understand performance may be impacted by this.

I proceeded to add each camera to Frigate, and at ~4 camera's, it max's the CPU of the guest, even if I give the guest 32 cores, and update the config to use multiple threads. 

I then created a container per camera, all on the same ubuntu guest referenced above, and all camera's have detection enabled, and are running without issue, so I would expect one container to be able to handle all the devices, with detection, given that the same platform is hosting multiple containers without issue. 

### Version

0.10.1-83481AF

### Frigate config file

```yaml
detectors:
  cpu1:
    type: cpu
  cpu2:
    type: cpu
  cpu3:
    type: cpu
  cpu4:
    type: cpu

mqtt:
  host: <removed for web posting>
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: user01
  password: nottherealpassword
  stats_interval: 60

ffmpeg:
  input_args:
    - -avoid_negative_ts
    - make_zero
    - -fflags
    - nobuffer
    - -flags
    - low_delay
    - -strict
    - experimental
    - -fflags
    - +genpts+discardcorrupt
    - -vsync
    - drop
    - -use_wallclock_as_timestamps
    - '1'

objects:
  track:
    - person
    - dog
  filters:
    person:
      min_area: 5000
      max_area: 100000
      min_score: 0.5
      threshold: 0.7
detect:
  width: 1920
  height: 1080
  fps: 5
record:
  enabled: True
  retain_days: 3
  events:
    retain:
      default: 10
snapshots:
  enabled: True
  timestamp: True
  bounding_box: False
  crop: False
  height: 175
  retain:
    default: 10
    objects:
      person: 15  
      dog: 15  

  
cameras:
  front-door:
    ffmpeg:
      inputs:
        - path: rtsp://<removed for web posting>@<removed for web posting>/live
          roles:
            - detect
            - rtmp
            - record
    motion:
      mask:
        - 0,0,317,0,1920,0,402,0,399,67,365,208,0,369
    zones:
      entry:
        coordinates: 0,295,0,1080,1920,1080,1920,0,399,0,421,95,243,171
  back-yard:
    ffmpeg:
      inputs:
        - path: rtsp://<removed for web posting>@<removed for web posting>/live
          roles:
            - -detect
            - rtmp
            - record
    motion:
      mask:
        - 0,0,317,0,1920,0,402,0,399,67,365,208,0,369
    zones:
      entry:
        coordinates: 0,295,0,1080,1920,1080,1920,0,399,0,421,95,243,171  
  living-room:
    ffmpeg:
      inputs:
        - path: rtsp://<removed for web posting>@<removed for web posting>/live
          roles:
            - -detect
            - rtmp
            - record
    motion:
      mask:
        - 0,0,317,0,1920,0,402,0,399,67,365,208,0,369
    zones:
      entry:
        coordinates: 0,295,0,1080,1920,1080,1920,0,399,0,421,95,243,171
  reptile-room:
    ffmpeg:
      inputs:
        - path: rtsp://<removed for web posting>@<removed for web posting>/live
          roles:
            - -detect
            - rtmp
            - record
    motion:
      mask:
        - 0,0,317,0,1920,0,402,0,399,67,365,208,0,369
    zones:
      entry:
        coordinates: 0,295,0,1080,1920,1080,1920,0,399,0,421,95,243,171
```

### Relevant log output

```shell
Will reconfigure container01 and update with them
```

### FFprobe output from your camera

```shell
Will reconfigure container01 and update with them
```

### Frigate stats

```json
{"detection_fps":0.0,"detectors":{"cpu1":{"detection_start":0.0,"inference_speed":126.69,"pid":242}},"front-door":{"camera_fps":20.0,"capture_pid":249,"detection_fps":0.0,"pid":246,"process_fps":20.0,"skipped_fps":0.0},"service":{"storage":{"/dev/shm":{"free":63.4,"mount_type":"tmpfs","total":67.1,"used":3.7},"/media/frigate/clips":{"free":24111.1,"mount_type":"ext4","total":41478.8,"used":15230.3},"/media/frigate/recordings":{"free":24111.1,"mount_type":"ext4","total":41478.8,"used":15230.3},"/tmp/cache":{"free":839.8,"mount_type":"tmpfs","total":1000.0,"used":160.2}},"temperatures":{},"uptime":1356,"version":"0.10.1-83481af"}}
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wireless

### Camera make and model

Wyze v2 & v3

**Comment by NickM-27** [comment]
> I am not using a coral device [...] I understand performance may be impacted by this.

To be clear, performance of a coral is a lot better than even the best CPUs, so its definitely a contributor. 

As far as the actual issue goes, I find it interesting that multiple containers performs better than in one, this makes me think it may be related to CPU setup for detectors. It would be better to know for sure though, if you are able to get them into a single container and run `top` it would be helpful to see which process is using resources.

I'd check out https://docs.frigate.video/configuration/index with the detectors section, might be worthwhile defining num_threads and playing with CPU count as well.

**Comment by Narolth** [comment]
Thanks NickM-27, 

After a lot more testing, I do see less performance with multiple containers on the same guest, than I had thought, I just needed to let them run much longer before having issues. 

I have a reasonably busy week this week, but I'll grab some logs, and review your suggestions and update asap, thanks again!

**Comment by Narolth** [comment]
I rebuilt my docker guest, to use CentOS 7, I had been using Ubuntu, and there is a large difference in performance. I have 6 docker instances of Frigate running, 3 are using motion detection / object detection. and I gave the 8 vCPU's, and  16 gb of memory. 

I still see issues when I load more than a couple of camera's onto an instance, specifically 1 with object detection, and 2 or more with out. 

Taking a look at the https://docs.frigate.video/configuration/index next.
